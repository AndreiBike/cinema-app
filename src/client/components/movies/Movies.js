import React from 'react';
import styles from './Movies.module.css';
import MovieContainer from './Movie/MovieContainer';


const Movies = (props) => {

  if (props.movies.length === 0) {
    return (
      <div className={styles.emptyMovies}>
        <span>No films found </span>
      </div>
    )
  }
  else {

    let moviesList = props.movies.map((movie)=> {
      return (
        <MovieContainer key = {movie.id} movie= {movie} />
      )
    })

    return (
      <div className={styles.movies}>
        {moviesList}
      </div>
    )
  }
}

export default Movies;
