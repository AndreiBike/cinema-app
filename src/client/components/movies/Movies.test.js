import React from 'react';
import { configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Movies from './Movies';

configure({ adapter: new Adapter() });

describe("<Movies /> component testing", () => {
  const uploadMovies =jest.fn()
  let component = mount(<Movies movies={[]}
    isLoading={false}
    sortBy={'release_date'}
    searchText={''}
    searchBy={'title'}
    offset={0}
    total={0}
    uploadMovies = {uploadMovies}
  />)

  it("+++capturing snapshot of Movies ", () => {
    expect(component).toMatchSnapshot();
  });
});