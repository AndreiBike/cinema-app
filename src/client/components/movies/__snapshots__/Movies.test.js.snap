// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Movies /> component testing +++capturing snapshot of Movies  1`] = `ReactWrapper {}`;
